File: nodejs/checkio/src/initiation/correct_sentence.ts
import assert from "assert";

function correctSentence(text: string): string {
	const start = text[0].toUpperCase();
	if (!text.endsWith(".")) {
		text += ".";
	}
	return start + text.slice(1);
}

console.log("Example:");
console.log(correctSentence("greetings, friends"));

// These "asserts" are used for self-checking
assert.strictEqual(
	correctSentence("greetings, friends"),
	"Greetings, friends.",
);
assert.strictEqual(
	correctSentence("Greetings, friends"),
	"Greetings, friends.",
);
assert.strictEqual(
	correctSentence("Greetings, friends."),
	"Greetings, friends.",
);
assert.strictEqual(
	correctSentence("greetings, friends."),
	"Greetings, friends.",
);

console.log("Coding complete? Click 'Check Solution' to earn rewards!");

File: python/checkio/initiation/correct-sentence.py
def correct_sentence(text: str) -> str:
    if text[-1] != ".":
        text += "."
    if text[0].islower():
        text = text[0].upper() + text[1:]
    return text


print("Example:")
print(correct_sentence("greetings, friends"))

# These "asserts" are used for self-checking
assert correct_sentence("greetings, friends") == "Greetings, friends."
assert correct_sentence("Greetings, friends") == "Greetings, friends."
assert correct_sentence("Greetings, friends.") == "Greetings, friends."
assert correct_sentence("greetings, friends.") == "Greetings, friends."

print("The mission is done! Click 'Check Solution' to earn rewards!")
